code/modules/mining/machine_vending.dm

```diff
@@ -93,13 +93,18 @@
 	)
 	prize_list["Extra"] = list() // Used in child vendors
 
+/obj/machinery/mineral/equipment_vendor/proc/remove_id()
+	if(inserted_id)
+		inserted_id.forceMove(get_turf(src))
+		inserted_id = null
+		return TRUE
+
 /obj/machinery/mineral/equipment_vendor/power_change()
-	if(!..())
-		return
+	. = ..()
 	update_icon(UPDATE_ICON_STATE)
 	if(inserted_id && !(stat & NOPOWER))
```

Comment:
```suggestion
	if(. && inserted_id && (stat & NOPOWER))
```
This check should work
if `stat has changed` then check to see if there is NOPOWER, if true then spit it out